backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/document/model/dsl/impl/ElasticsearchIndexSchemaElementImpl.java

```diff
@@ -76,4 +83,13 @@ public ElasticsearchIndexSchemaObjectField objectField(String relativeFieldName,
 				} );
 	}
 
+	private static void checkRelativeFieldName(String relativeFieldName) {
+		if ( relativeFieldName == null || relativeFieldName.isEmpty() ) {
+			throw log.relativeFieldNameCannotBeNullOrEmpty( relativeFieldName );
+		}
+		if ( relativeFieldName.contains( "." ) ) {
+			throw log.relativeFieldNameCannotContainDot( relativeFieldName );
+		}
+	}
```

Comment:
Wondering if this restriction shouldn't be in the engine itself. I don't think it's a bad thing to generalize it and be consistent across backends.

Reply:
I didn't for several reasons:

1. We might want to execute this check in code where the engine is not even involved, for instance when adding a backend-specific field from a bridge.
2. We might want, for some backends, to be *more* restrictive, for instance disallow non-alphanumeric characters. I'll have to check but I doubt Elasticsearch allows just any character, there are probably some non-alphanumeric characters that are forbidden.
3. That's a bit more far-fecthed, but we might want, for some backends, to be *less* restrictive. I think Elasticsearch supports dots in field names from version 5 onward. I don't know how we would handle that when building search predicates, though.

I guess we could solve 1 & 2 by passing an object responsible for the "main" checks from the engine to the backend, and then expecting the backend to use it whenever a field is added. But I'd rather handle that in another ticket... I created [HSEARCH-3152](https://hibernate.atlassian.net/browse/HSEARCH-3152) and will do what I can when I'll address that ticket.

Reply:
Keep in mind that our goal is also to provide a somewhat consistent experience across our backends.

I'm not sure it's really worth it to consider allowing special characters for one backend and not the others. I would choose the common denominator and that's it. If you can't use `§` in your field name on a backend that allows it, so be it.

And sure we can make it another issue.